### PR TITLE
field cache is refreshed when loading discover, visualize,or analyze tabs

### PR DESCRIPTION
**Pull request by alexweltman**

### CODE REVIEW

File: src/kibana/components/index_patterns/index_patterns.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ define(function (require) {
     var patternCache = Private(require('components/index_patterns/_pattern_cache'));
 
     var notify = new Notifier({ location: 'IndexPatterns Service'});
+    var DEFAULT_INDEX_PATTERN = '[network_]YYYY_MM_DD';
```
Comment (john-gress) [inline src/kibana/components/index_patterns/index_patterns.js]: Why deviate from the naming convention?

File: src/kibana/plugins/visualize/editor/editor.js [hunk 2]
```diff
@@ -48,9 +49,13 @@ define(function (require) {
...
     'kibana/notify',
-    'kibana/courier'
+    'kibana/courier',
+    'kibana/index_patterns',
   ])
-  .controller('VisEditor', function ($scope, $route, timefilter, AppState, $location, kbnUrl, $timeout, courier, Private, Promise) {
+  .controller('VisEditor', function ($scope, $route, timefilter, AppState, 
+     $location, kbnUrl, $timeout, courier, Private, Promise, indexPatterns) {
+     
+    indexPatterns.refresh();
```
Comment (KjellKod) [inline src/kibana/plugins/visualize/editor/editor.js]: @alexweltman where's the button?

Comment (KjellKod) [inline src/kibana/plugins/visualize/editor/editor.js]: ### just kidding

File: src/kibana/components/index_patterns/index_patterns.js (hunk 1 above)
Comment (sammccord) [inline src/kibana/components/index_patterns/index_patterns.js]: All uppers is pretty common for constants. We're not using ES6 javascript yet so there's no built-in way to distinguish between constants and variables.

**Comment by sammccord** [comment]
👍

**Comment by john-gress** [comment]
👍

**Comment by craig-cogdill** [comment]
👍  Good stuff

**Comment by KjellKod** [comment]
👍 tripple verified

**Pull request merged**